cli/src/feature.rs

```diff
@@ -487,47 +583,28 @@ fn feature_activation_allowed(rpc_client: &RpcClient, quiet: bool) -> Result<boo
             max_stake_percent_len = max_stake_percent_len.max(stake_percent.len());
             max_rpc_percent_len = max_rpc_percent_len.max(rpc_percent.len());
 
-            stats_output.push((
+            feature_sets.push(CliFeatureSet {
                 software_versions,
                 feature_set,
                 stake_percent,
                 rpc_percent,
                 me,
-            ));
-        }
-        println!(
-            "{}",
-            style(format!(
-                "{1:<0$}  {3:<2$}  {5:<4$}  {7:<6$}",
-                max_software_versions_len,
-                software_versions_title,
-                max_feature_set_len,
-                feature_set_title,
-                max_stake_percent_len,
-                stake_percent_title,
-                max_rpc_percent_len,
-                rpc_percent_title,
-            ))
-            .bold(),
-        );
-        for (software_versions, feature_set, stake_percent, rpc_percent, me) in stats_output {
-            println!(
-                "{1:<0$}  {3:>2$}  {5:>4$} {7:>6$}  {8}",
-                max_software_versions_len,
-                software_versions,
-                max_feature_set_len,
-                feature_set,
-                max_stake_percent_len,
-                stake_percent,
-                max_rpc_percent_len,
-                rpc_percent,
-                if me { "<-- me" } else { "" },
-            );
+            });
         }
-        println!();
-    }
+        Some(CliClusterFeatureSets {
+            tool_feature_set: my_feature_set,
+            feature_sets,
+            tool_feature_set_matches_cluster,
+            stake_allowed,
+            rpc_allowed,
+            max_software_versions_len,
+            max_feature_set_len,
+            max_stake_percent_len,
+            max_rpc_percent_len,
```

Comment:
nit: I think it makes more sense to recompute these in the display implementation since they're only used there and it would help keep the `CliClusterFeatureSets` type more focused

Reply:
Mm, yeah, I'd like to avoid an extra iteration through the feature sets, but maybe I can find a way to move the one in `feature_activation_allowed()` into the display implementation completely.

Reply:
Done!